File: Common/Actor/src/common/loaders/ImageLoader.java
/**
 * @author Julian Sender
 */
package common.loaders;

import javafx.embed.swing.SwingFXUtils;
import javafx.scene.image.Image;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.*;

/**
 * loading image
 */
public final class ImageLoader {

    /**
     * Returns Gets a path and returns image
     * @param relativeFilePath the relative path to the image
     * @return returns imagepath for SwingFXUtils
     */
    public static Image loadImage( final String relativeFilePath ) throws IOException {
        try(InputStream u = getInputStream( relativeFilePath )) {
            BufferedImage image = ImageIO.read( u );
            assert image != null;
            return SwingFXUtils.toFXImage( image, null );
        }
    }

    /**
     *
     * Gets Inputstream of resources as Stream for pictures depending on relative path to picture
     * @param relativePath relative path to the picture
     * @return returns an inputstream containing the image
     * @throws NullPointerException Location not found
     */
    private static InputStream getInputStream( String relativePath ) throws NullPointerException,
            FileNotFoundException {
        java.net.URL x = ImageLoader.class.getResource(relativePath);
        if (x == null) {
            throw new FileNotFoundException("File does not exist under relative Path: " + relativePath);
        }
        File f = new File(x.getPath());
        if (!f.exists()) {
            throw new FileNotFoundException("File does not exist: " + x.getPath());
        }
        return ImageLoader.class.getResourceAsStream(relativePath);
    }

}

File: Common/Actor/src/common/actor/ImageLoader.kt
/**
 * @author Julian Sender
 */
package common.actor

import javafx.embed.swing.SwingFXUtils
import javafx.scene.image.Image
import java.io.File
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStream
import javax.imageio.ImageIO

/**
 * loading image
 */
object ImageLoader {
    /**
     * Returns Gets a path and returns image
     * @param relativeFilePath the relative path to the image
     * @return returns imagepath for SwingFXUtils
     */
    @JvmStatic
    @Throws(IOException::class)
    fun loadImage(relativeFilePath: String): Image =
            getInputStream(relativeFilePath).use { u ->
                return SwingFXUtils.toFXImage(ImageIO.read(u)!!, null)
            }

    /**
     *
     * Gets Inputstream of resources as Stream for pictures depending on relative path to picture
     * @param relativePath relative path to the picture
     * @return returns an inputstream containing the image
     * @throws NullPointerException Location not found
     */
    @Throws(NullPointerException::class, FileNotFoundException::class)
    private fun getInputStream(relativePath: String): InputStream {
        val x = ImageLoader::class.java.getResource(relativePath)
                ?: throw FileNotFoundException("File does not exist under relative Path: $relativePath")
        if (!File(x.path).exists()) {
            throw FileNotFoundException("File does not exist: " + x.path)
        }
        return ImageLoader::class.java.getResourceAsStream(relativePath)
    }
}
